provisioning/wildfly_channel_in_zips.adoc

```diff
@@ -0,0 +1,210 @@
+However for users that are downloading WildFly from our traditional distributions (zip & tgz archives) they can not take advantage of this mechanism.
+
+If WildFly archives would work out of the box with channels, it would significantely simplify updating an installation.
+
+The workflow would be as simple as:
+
+1. Download WildFly version x.y.z from our download page
+2. Use it as normal
+3. A more recent version of WildFly x.y.(z+ 1) is made available
+4. Run a script in the existing WildFly installation to update it to this new version
+5. Restart the server to run your applications on this new version
+
+If there are multiple available updates for WildFly, the user must be able to decide which one they want to update to.
+This lets the user decide whether they want to update to the latest micro release or the next major release.
+
+=== WildFly Channels
+
+WildFly channels are published for each of its distributions since 32.0.0.Beta1:
+
+* https://repo1.maven.org/maven2/org/wildfly/channels/wildfly/[org.wildfly.channels:wildfly] 
+* https://repo1.maven.org/maven2/org/wildfly/channels/wildfly-ee/[org.wildfly.channels:wildfly-ee] 
+* https://repo1.maven.org/maven2/org/wildfly/channels/wildfly-preview/[org.wildfly.channels:wildfly-preview]
+
+These channels are versioned based on the WildFly versions (e.g. `org.wildfly.channels:wildfly:32.0.1.Final` contains all the dependencies to provision WildFly 32.0.1.Final's full installation)
+
+The http://docs.wildfly.org/wildfly-proposals/build/WFLY-19130_publish_Wildfly_channel_manifest.html[Analysis Document for WFLY-19130] states that the channels publish the latest manifest with no backwards compatibility guarantee, meaning each channel will always contain components of the latest released version of WildFly.
+
+There are no guarantee that the channel will not break between major releases of WildFly.
+We guarantee that minor and micro releases of WildFly do not contain breaking changes and this extends to their channels.
+
+=== Tooling Workflow
+
+With that in mind, we propose to incorporate the channel metadata in the WildFly distributions that describes the exact version of the channel used to provision the server.
+In practice, that means that the installation would be "bound" to a versioned channel (eg `org.wildfly.channels:wildfly:33.0.0.Final`).
+
+Updating to a more recent version is a user decision and would require the user to give as input the version of the WildFly they want to _update to_ (e.g. `33.0.1.Final` or `34.0.0.Final`).
+After the update is successful, the channel metadata would be updated to point to the current version of the installation.
+
+WildFly must contain the tooling to let user update installations without having to download another piece of software.
+
+The tooling should be made available in the `bin` directory of this installation and affects only the parent directory (`JBOSS_HOME` would have no bearing for update).
```

Comment:
We should keep JBoss CLI and HAL in sync with those new capabilities.

Reply:
FYI on tooling alignment (especially HAL and JBoss CLI) I have a discussion here about a change I will be proposing against the proposals repo.

Overall based on the stability level of a new feature I am proposing different levels of support in the tooling so features can come it at a lower level maybe without tooling support and then get the tooling support added, then promote the feature to a higher stability level.

A general exception however would be where a feature is dependent on the tooling support entierely, i.e. if you can not use it without the JBoss CLI or HAL then one must support it at the outset - assuming the developers on those projects are available to assist.